### DISCUSSION TITLE
Using later versions on InfluxDB

### DISCUSSION THREAD

**Discussion by Chrisdup8710** [ptarmiganlabs/butler-sos#803] [Ideas]
Good day

Do we know if ButlerSOS would work on later versions of InfluxDB? At our client we are being flagged for obsolescence on the InfluxDB v1 we are using for Butler.

**Comment by mountaindude** [comment]
_Edit: 
Text below mentions Butler, rather than Butler SOS. 
The need for adding support for later InfluxDB versions is the same for both products.
I just wasn't paying attention to which repository the post above was created in... Sorry :/_

-----



Ah, good question.

Butler does currently ONLY work on InfluxDB 1.x.

That version is quite old by now, no question about that.

Adding support for later InfluxDB version has been a request for years,
never really been prioritized though.

So... we should just get it done for both Butler and Butler SOS, as those
tools are usually used in tandem.
I don't have a time plan for this though (yet)

**Comment by mountaindude** [comment]
@Chrisdup8710
I'll start poking around this idea, without any specific deadlines though.

Would you guys be able to beta test a coming Butler SOS version that supports InfluxDB 2.x?
It's a pretty significant change that needs to happen, so it's good to get more testers before the new version goes public.

**Comment by mountaindude** [comment]
Btw, here is an interesting article about the future of InfluxDB. Very promising and confirms it's time to upgrade from v 1.x.
https://www.influxdata.com/blog/the-plan-for-influxdb-3-0-open-source/

**Comment by mountaindude** [comment]
@Chrisdup8710 

Butler SOS 10.1.0 was released earlier today, adding support for InfluxDB 2.x.
It's a rather significant update with major changes to handle both InfluxDB 1.x and 2.x

Give it a spin!

**Comment by mountaindude** [comment]
Download page here: https://github.com/ptarmiganlabs/butler-sos/releases
